File: Domain/Entities/Category.cs
using Domain.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities
{
	public class Category : WithId, IWithUserId, IWithSrc
	{
		public string Title { get; set; }
		public string Description { get; set; }

		public string Src { get; set; }

		public List<Item> Items { get; set; }

		[ForeignKey(nameof(UserId))]
		public User User { get; set; }
		public Guid UserId { get; set; }
	}
}

File: WebApi/ClientApp/src/objects/Category.ts
import { Item } from "./Item";
import { CategoryType } from "./enums";

export interface Category {
	id: string;
	title: string;
	description: string;
	categoryType: CategoryType;
	items: Array<Item>;
	itemsCount: number;
	src: string;
}

export const CategotyTypeNames = {
	[CategoryType.None]: 'None',
	[CategoryType.Games]: 'Games',
	[CategoryType.Films]: 'Films',
	[CategoryType.Books]: 'Books',
}